test/built-ins/Iterator/from/return-method-calls-base-return-method.js

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2024 Sosuke Suzuki. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-iterator.from
+description: >
+  %WrapForValidIteratorPrototype%.return() call base iterator's return method when it exists.
+info: |
+  %WrapForValidIteratorPrototype%.return ( )
+    5. Let returnMethod be ? GetMethod(iterator, "return").
+    6. If returnMethod is undefined, then
+      ...
+    7. Return ? Call(returnMethod, iterator).
+
+includes: [deepEqual.js]
+features: [iterator-helpers]
+flags: []
+---*/
+
+let returnCallCount = 0;
+const iter = {
+    return () {
+        returnCallCount++;
+        return { value: 5, done: true };
+    },
+};
+const wrapper = Iterator.from(iter);
+
+assert.deepEqual(wrapper.return(), { value: 5, done: true });
```

Comment:
We are trying not to add new uses of `assert.deepEqual` (see #3476). One thing you could do instead in this case, is compare by reference:
```js
const expectedIteratorResult = { value: 5, done: true };
...
const iter = {
  return() {
    returnCallCount++;
    return expectedIteratorResult;
  },
};
...
assert.sameValue(wrapper.return(), expectedIteratorResult);
```